src/hazelcore/OcamlExtraction.re

```diff
@@ -0,0 +1,374 @@
+module Typ = {
```

Comment:
let's put all of the extraction modules into a directory `hazelcore/extractions/ocaml`

Reply:
also you should separate these out into OCamlExtraction_Typ, OCamlExtraction_Exp, etc. following the recent refactoring

Reply:
this wasn't resolved